src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,811 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
-                    if(Advancement_Info.useAdv() && Advancement_Info.getApiHEDA() != null) {
-                        Advancement_Info.getApiHEDA().primaryContact(acc, acc.contacts[0].Id);
-                    }
+    /*******************************************************************************************************
+    * @description Processes after delete logic.
+    * @param oldContacts is a list of Contacts from trigger old.
+    * @param accountId is the AccountId of Contact.
+    */
+    private void handlesAfterDelete(List<Contact> oldContacts, Id accountId) {
+        List<Contact> contactAccountDelete = new List<Contact>();
+        List<Account> relatedAccounts = new List<Account>();
+        Set<Id> accountIdsToRename = new Set<Id>();
+        Set<Id> relatedAccountIds = new Set<Id>();
+
+        Integer numOfContactsLeft = checkContactState(accountId);
+        Id accRecordTypeId = returnAccountRecordTypeId(accountId);
+
+        for (Contact c : oldContacts) {
+            if (numOfContactsLeft == 0) {
+                if (checkHouseholdNaming(accRecordTypeId)) {
+                    if (c.AccountId != accountId) {
+                    	contactAccountDelete.add(c);
+                    	relatedAccountIds.add(c.AccountId);
+                	}
                 } else {
-                    //if no child contact exists, clear up primary contact
-                    acc.Primary_Contact__c = null;
+                    contactAccountDelete.add(c);
+                	relatedAccountIds.add(c.AccountId);
                 }
-                listAccToUpdate.add(acc);
+            } else {
+                contactAccountDelete.add(c);
+                relatedAccountIds.add(c.AccountId);
             }
+
         }
 
-        // now do the update
-        dmlWrapper.objectsToUpdate.addAll((List<SObject>)listAccToUpdate);
+        //Requery to retrieve correct Account value
+        relatedAccounts = [SELECT Id, RecordTypeID
+                           FROM Account
+                           WHERE Id IN :relatedAccountIds];
+
+        for (Account account : relatedAccounts) {
+            if (checkHouseholdNaming(account.RecordTypeID)) {
+                accountIdsToRename.add(account.Id);
+            }
+        }
+
+        handleDeleteProcessing(accountIdsToRename, contactAccountDelete);
     }
 
     /*******************************************************************************************************
-    * @description Inserts a new Account for an Individual Contact
-    * @param contacts the list of Contacts that need updating.
-    * @param dmlWrapper to hold the Contacts that need updating.
-    * @return void
+    * @description This method retrieves the recordTypeId of the Account.
+  	* @param accountId is the Id of the Account.
+	* @return Id.
     */
-    private void insertContactAccount(Contact[] contacts, DmlWrapper dmlWrapper) {
+    private Id returnAccountRecordTypeId(Id accountId){
```

Comment:
@lifewithryan , it looks like returnAccountRecordTypeId() method was created so that we can use it in the handlesAfterDelete() method on line 453. The only place I can see to retrieve the Account RecordTypeId is before handleAfterDelete() method is called on line 116.  This means I need to retrieve the Account RecordTypeId and create a separate method to query Contact's Account RecordTypeId for 'oldContacts' specifically since queryContacts() method which is called on line 94 is only for 'newContacts'.  Also, for handlesAfterDelete() method it is calling handleDeleteProcessing() method which is calling deleteContactAccountsIfEmpty() method and that's the only other one in this class that utilizes Account RecordTypeId but it's only comparing it is null or not. The other Account RecordTypeId comparison you're finding might be for handleInsertProcessing() and HandleUpdateWrapperLogic() which is pulling the Account recordtypeId from queryContacts() methods.

Reply:
cool no changes needed :)